src/spartan/ppsnark.rs

```diff
@@ -580,7 +580,12 @@ impl<E: Engine, EE: EvaluationEngineTrait<E>> RelaxedR1CSSNARKTrait<E> for Relax
     let poly_vec = vec![&Az, &Bz, &Cz];
     let c = transcript.squeeze(b"c")?;
     let w: PolyEvalWitness<E> = PolyEvalWitness::batch(&poly_vec, &c);
-    let u: PolyEvalInstance<E> = PolyEvalInstance::batch(&comm_vec, &tau_coords, &eval_vec, &c);
+    let u: PolyEvalInstance<E> = PolyEvalInstance::batch(
+      &comm_vec,
+      tau_coords[..comm_vec.len()].to_vec(),
```

Comment:
`tau_coords` should be copied as is, its size does not depend on the number of commitments.